spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/session/SessionAutoConfigurationJdbcTests.java

```diff
@@ -86,4 +88,15 @@ public void customTableName() {
 				.queryForList("select * from FOO_BAR")).isEmpty();
 	}
 
+	@Test
+	public void customTableNameWithDefaultSchemaFails() {
+		this.thrown.expect(BeanCreationException.class);
+		this.thrown.expectMessage(containsString(
+				"Database initializer cannot be configured"));
+		load(Arrays.asList(EmbeddedDataSourceConfiguration.class,
+				DataSourceTransactionManagerAutoConfiguration.class),
+				"spring.session.store-type=jdbc",
+				"spring.session.jdbc.table-name=FOO_BAR");
```

Comment:
This test doesn't feel right at all. So basically you're saying you want JDBC and a custom table name and you'll get an exception? I would rather move the initialization of the table conditional on whether a custom table name has been specified or not. That's consistent with other places where we determine a flag based on the environment.

You provided a custom table name? Then you're on your own to create the schema (instead of throwing an exception as this PR does). 

You provided the path to a script? We'll of course use that to initialize the database

You didn't provide anything? We're going to assume you want us to create the schema with the default table name.

That seems a much better solution to the problem. If you agree, can you rework the PR?